File: Blog.Core/Infrastructure/Pagination/SearchQuery.cs
namespace Blog.Core.Infrastructure.Pagination
{
    using Blog.Core.Infrastructure.Pagination.Interfaces;
    using System;
    using System.Collections.Generic;
    using System.Linq.Expressions;
    public class SearchQuery<TEntity>
    {
        public SearchQuery()
        {
            Filters = new List<Expression<Func<TEntity, bool>>>();
            SortCriterias = new List<ISortCriteria<TEntity>>();
        }

        public List<Expression<Func<TEntity, bool>>> Filters { get; protected set; }

        public void AddFilter(Expression<Func<TEntity, Boolean>> filter)
        {
            Filters.Add(filter);
        }

        public List<ISortCriteria<TEntity>> SortCriterias
        {
            get;
            protected set;
        }

        public void AddSortCriteria(ISortCriteria<TEntity> sortCriteria)
        {
            SortCriterias.Add(sortCriteria);
        }

        public string IncludeProperties { get; set; }

        public int Skip { get; set; }

        public int Take { get; set; }
    }
}

File: Shared/Blog.FSharp.Core/Infrastructure/Pagination/SearchQuery.fs
namespace Blog.FSharp.Core.Infrastructure.Pagination

open System
open System.Collections.Generic
open System.Linq.Expressions
open Blog.FSharp.Core.Infrastructure.Pagination.Interfaces

/// <summary>
/// Search query.
/// </summary>
/// <typeparam name="TEntity">TEntity.</typeparam>
type SearchQuery<'TEntity>() =

    let filters = List<Expression<Func<'TEntity, bool>>>()
    let sortCriterias = List<ISortCriteria<'TEntity>>()

    /// <summary>
    /// Gets or sets sort criterias.
    /// </summary>
    member val SortCriterias : List<ISortCriteria<'TEntity>> = sortCriterias with get, set

    /// <summary>
    /// Gets or sets include properties.
    /// </summary>
    member val IncludeProperties : string = null with get, set

    /// <summary>
    /// Gets or sets skip.
    /// </summary>
    member val Skip = 0 with get, set

    /// <summary>
    /// Gets or sets take.
    /// </summary>
    member val Take = 0 with get, set

    /// <summary>
    /// Gets or sets filters.
    /// </summary>
    member val Filters : List<Expression<Func<'TEntity, bool>>> = filters with get, set

    /// <summary>
    /// Add filter.
    /// </summary>
    member this.AddFilter(filter: Expression<Func<'TEntity, bool>>) =
        this.Filters.Add(filter)

    /// <summary>
    /// Add sort criteria params.
    /// </summary>
    member this.AddSortCriteria(sortCriteria: ISortCriteria<'TEntity>) =
        this.SortCriterias.Add(sortCriteria)